presto-verifier/src/main/java/com/facebook/presto/verifier/framework/AbstractVerification.java

```diff
@@ -276,10 +286,25 @@ private EventStatus getEventStatus(
         if (resolveMessage.isPresent()) {
             return FAILED_RESOLVED;
         }
-        if ((skipControl && testQueryContext.getState() == QueryState.SUCCEEDED)
-                || (!skipControl && matchResult.isPresent() && matchResult.get().isMatched())) {
-            return SUCCEEDED;
+
+        if (skipControl) {
+            if (testQueryContext.getState() == QueryState.SUCCEEDED) {
+                return SUCCEEDED;
+            }
+        }
+        else {
+            if (skipChecksum) {
+                if (controlQueryContext.getState() == QueryState.SUCCEEDED && testQueryContext.getState() == QueryState.SUCCEEDED) {
+                    return SUCCEEDED;
+                }
+            }
+            else {
+                if (matchResult.isPresent() && matchResult.get().isMatched()) {
+                    return SUCCEEDED;
+                }
+            }
```

Comment:
A nit would be to try and simplify this to remove all the non-logical things from the if statements and pull out the facts (did X succeed) into a series of booleans.

So the big messy if predicates would be all variable names that describe the predicate more explicitly.